Add support for PHP 8.1.0

mkoprek opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description
Add support for PHP 8.1.0 for laminas/laminas-diagnostics

Drop support for PHP 7.3

Cause there is no any tag yet in **laminas/laminas-code** I used a commit hash

Fixes #30

mkoprek committed:
Add support for PHP8.1

Signed-off-by: Maciej Koprek <mkoprek@gmail.com>

samsonasik commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
+.idea/
```
please revert this for concistency, you can add yours in your own system global gitignore

mkoprek replied:
Removed

sasezaki commented on src/Check/Memcached.php:
```diff
@@ -72,7 +72,7 @@ public function check()
             ) {
                 // Attempt a connection to make sure that the server is really down
-                if (! @$memcached->getLastDisconnectedServer($this->host, $this->port)) {
+                if ($memcached->getLastErrorCode() !== 0) {
```
Why changed ?

mkoprek replied:
@sasezaki There was a error that getLastDisconnectedServer() need to have exact 0 argument, but has 2

Ocramius replied:
Hmm, this code path doesn't seem to be tested at all: we can't change this without an integration test :thinking:

mkoprek replied:
@Ocramius Ok, so I should go back to previous version and do some if for PHP version?

mkoprek replied:
@Ocramius $memcached->getLastDisconnectedServer return array if there was any error.I changed how it should be handled in here and removed my proposition with getLastErrorCode().

mkoprek committed:
Remove .idea from code

Signed-off-by: Maciej Koprek <mkoprek@gmail.com>

Ocramius commented on composer.json:
```diff
@@ -27,13 +27,14 @@
         "doctrine/migrations": "^2.0 || ^3.0",
         "enlightn/security-checker": "^1.2",
         "guzzlehttp/guzzle": "^5.3.3 || ^6.3.3",
+        "laminas/laminas-code": "4.5.x-dev#fbdaecda0c91ea0c7aa81c89eda16b2f345d0657 as 4.5.0",
```
This should be dropped before merging

mkoprek replied:
@Ocramius You mean whole line, or just commit sha? The problem is that laminas-code 4.4.* is not supporting PHP 8.1 and 4.5 tag is not pushed yet.

Ocramius replied:
Correct - whole line should be removed. Well aware that `laminas-code` does not support `php: 8.1` yet.

mkoprek replied:
Done

mkoprek committed:
Merge 1.11.x and revert changes in Memcached.php

mkoprek committed:
Rollback changes to $memcached->getLastDisconnectedServer()

Signed-off-by: Maciej Koprek <mkoprek@gmail.com>

mkoprek committed:
Remove laminas/laminas-code sticked to commit SHA

Signed-off-by: Maciej Koprek <mkoprek@gmail.com>

mkoprek commented:
I removed locking laminas/laminas-code in composer.json
Also I changed if with $memcached->getLastDisconnectedServer(). 

Now all should be good, but there will be problem with composer install until laminas/laminas-code 4.5.0 will be released.

mkoprek committed:
Update laminas/laminas-code to 4.5.0

Signed-off-by: Maciej Koprek <mkoprek@gmail.com>

mkoprek commented:
@Ocramius I changes things you mentioned, could you please review it?

Ocramius reviewed (approved):
LGTM :+1: 

Thanks @mkoprek!
